pkg/util/kubernetes/apiserver/hpa_controller.go

```diff
@@ -34,6 +35,8 @@ import (
 const (
 	// maxRetries is the maximum number of times we try to process an autoscaler before it is dropped out of the queue.
 	maxRetries = 10
+	// maxMetricsCount is the maximum number of metrics we can query from the backend.
+	maxMetricsCount = 45
```

Comment:
do we wanna make it configurable, for users who have higher rate limits, or is it unrelated?

Reply:
it's unrelated indeed, this is how big a query can be, if it's > 55 metrics it will trigger the issue, that's why I don't want to have it configurable.

Reply:
is it 45 or 55? 🤔